public_html/wp-content/mu-plugins/blocks/source/blocks/schedule/schedule-grid.js

```diff
@@ -0,0 +1,331 @@
+ *          https://github.com/WordPress/gutenberg/issues/8118#issuecomment-457843428
+ *
+ * add proptypes everywhere
+ * review any other things necessary to apply changes from https://github.com/WordPress/wordcamp.org/commit/92aeab060a26faccf6c7267dc2520a166a8dacc6
+ *
+ *
+ * patch fixes #3117, #3842
+ * props mark, mel, others
+ */
+
+/**
+ * Render the schedule for a specific day.
+ *
+ * @param {string} date In a format acceptable by `wp.date.format()`.
+ * @param {Array} sessions
+ * @param {Array} tracks
+ *
+ * @return {Element}
+ */
+function ScheduleDay( { date, sessions, tracks } ) {
+	return (
+		<Fragment>
+			<h2 className="wordcamp-schedule-date">
+				{/* todo the S token isn't working: https://github.com/WordPress/gutenberg/issues/15221
+				should use the site's format instead of hardcoding
+				*/}
+				{ format( 'l, F jS', date ) }
+			</h2>
+			{/* this needs to be editable, should also be a separate Heading block. so when inserting a schedule block,  */}
+			{/*We can make the text editable, though, with a reasonable default. If they remove the text, then we can automatically remove the corresponding h2 tag, to avoid leaving an artifact behind that affects margins/etc.*/}
+
+			<section id={ `wordcamp-schedule-day-${ format( 'Y-m-d', date ) }` } className="wordcamp-schedule-day">    {/* todo any other classes? */}
+				<GridColumnHeaders tracks={ tracks } />
+				<Sessions sessions={ sessions } />
+			</section>
+		</Fragment>
+	);
+}
+
+/**
+ * Render the column headers for a schedule.
+ *
+ * @param {Array} tracks
+ *
+ * @return {Element}
+ */
+function GridColumnHeaders( { tracks } ) {
+	return (
+		<Fragment>
+			{/* todo should these have a more semantic thing equivalent to <th> ?
+			maybe rename wordcamp-schedule-track-slot to wordcamp-schedule-track-heading?
+
+			// todo if possible, make sure not using inline styles anywhere, b/c they're hard to override
+				// ok to use it to set ids/classes, but don't want to set things that people would want to change
+				// grid column/row is probably fine, and unavoidable, but think about
+			*/}
+
+			<span className="wordcamp-schedule-column-header" aria-hidden="true" style={ { gridColumn: 'times' } }>
+				{ _x( 'Time', 'table column heading', 'wordcamporg' ) }
+			</span>
+
+			{/* todo should not be showing Foo track on 17th b/c there aren't any sessinos assigned to it*/}
+
+			{ tracks.map( ( track ) => (
+				<span
+					key={ track.id }
+					className="wordcamp-schedule-column-header"
+					aria-hidden="true"
+					style={ { gridColumn: `wordcamp-schedule-track-${ track.id }` } }
+				>
+					{ track.name }
+				</span>
+			) ) }
+
+			{/* todo test w/ 3+ tracks, make sure column headers match assigned sessions */}
+			{/* maybe different name because needs to be unique to avoid conflicting styles? */}
+			{/* need js to toggle aria-hidden when css changes display:none? is aria-hidden needed here? */}
```

Comment:
They're hidden on the mobile view, but appear on the desktop/Grid view. It seemed like it needed `aria-hidden` since it can change dynamically, but maybe that's not really an issue in real life, except maybe tablets switching from landscape to portrait? What do you think?

Reply:
`display: none` will already hide these from screen reader users, so I think it's redundant here. The only reason you'd use `aria-hidden` would be to hide visible content from screen readers (like if it's already on the page in `screen-reader-text`, for example).

Reply:
Ah, that's a good point, now I'm not sure why; maybe because it's not _essential_ information, and can be implied, so it'd be more convenient for screen reader users?

Do you remember, @mrwweb?

Reply:
Here's the rationale:
- The track column "headings" are neither headings nor table headers. They are semantically unmoored both in markup and source order. They're essentially a nice-to-have feature for people who can see the layout. Therefore, screen readers neither need them nor could make sense of them.
- The track information is included with each session so it's accessible there. (It should never be `display:none`'d.)

The column "headings" are `display: none` on mobile because there's no easy or obvious place to put them, and, again, the information is provided directly on each session.

Reference: https://css-tricks.com/building-a-conference-schedule-with-css-grid/#article-header-id-6

Reply:
Ah, that makes sense to me. Thanks! 

What do you think, Kelly?

Reply:
Ok, I see now– since I couldn't get the schedule to load, I had to just rely on reading the code & that wasn't clear. As long as this info does appear in the session itself, using `aria-hidden` is fine.